recipe_scrapers/felixkitchen.py

```diff
@@ -0,0 +1,105 @@
+# mypy: allow-untyped-defs
+
+import warnings
+
+from ._abstract import AbstractScraper
+from ._utils import normalize_string
+
+MARK_SEPARATOR = " "
+INGREDIENT_SEPARATOR = "• "
+
+BUG_REPORT_LINK = "https://github.com/hhursev/recipe-scrapers/issues"
+
+
+class FieldNotProvidedByWebsiteWarning(Warning):
+    pass
+
+
+def field_not_provided_by_website_warning(host, field):
+    message = (
+        "{} doesn't support seem to support the {} field. "
+        "If you know this to be untrue for some recipe, please submit a bug report at {}"
+    )
+
+    warnings.warn(
+        message.format(host, field, BUG_REPORT_LINK),
+        category=FieldNotProvidedByWebsiteWarning,
+    )
+
+
+class FelixKitchen(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "felix.kitchen"
+
+    def author(self):
+        result = self.soup.select("span.author > a")
+        if not result:
+            return None
+        return result[0].text
+
+    def title(self):
+        found = self.soup.find("meta", {"property": "og:title"})
+        return normalize_string(found.get("content"))
+
+    def total_time(self):
+        field_not_provided_by_website_warning(self.host(), "total_time")
```

Comment:
Ok, yep - this is getting closer to a good interface I think.

I'd be tempted to suggest that in fact, this should be implemented as an exception.  So all that a new developer needs to know is: raise the correct type of exception.

```python
def total_time(self);
    raise FieldNotProvidedByWebsiteException()
```

...and then we'd have plugin -- enabled by default -- to emit a warning when that exception is raised.

That way, if some downstream consumer really wants to handle cases where the website doesn't _ever_ provide data for a field, they can do so.

(that's not possible -- or not easily possible -- using warnings alone, because they don't affect the return value/behaviour of the method)

Reply:
I'm going to merge this as-is, but I think we can and should refactor this in the nearish future.